libretroshare/src/pgp/pgphandler.cc

```diff
@@ -1404,35 +1406,38 @@ bool PGPHandler::SignDataBin(const RsPgpId& id,const void *data, const uint32_t
 	PGPFingerprintType fp(f.fingerprint) ;
 #endif
 
-    bool last_passwd_was_wrong = false ;
-ops_secret_key_t *secret_key = NULL ;
+	bool last_passwd_was_wrong = false ;
+	ops_secret_key_t *secret_key = NULL ;
 
-    for(int i=0;i<3;++i)
-    {
-        bool cancelled =false;
-        std::string passphrase = _passphrase_callback(NULL,reason.c_str(),uid_hint.c_str(),"Please enter passwd for encrypting your key : ",last_passwd_was_wrong,&cancelled) ;//TODO reason
+	for(int i=0;i<3;++i)
+	{
+		bool cancelled =false;
+		// Need to be outside of mutex to not block GUI.
+		std::string passphrase = _passphrase_callback(NULL,reason.c_str(),uid_hint.c_str(),"Please enter password for encrypting your key : ",last_passwd_was_wrong,&cancelled) ;//TODO reason
 
-        secret_key = ops_decrypt_secret_key_from_data(key,passphrase.c_str()) ;
+		secret_key = ops_decrypt_secret_key_from_data(key,passphrase.c_str()) ;
 
-        if(cancelled)
-        {
-            std::cerr << "Key entering cancelled" << std::endl;
-            return false ;
-        }
-        if(secret_key)
-            break ;
+		if(cancelled)
+		{
+			std::cerr << "Key entering cancelled" << std::endl;
+			return false ;
+		}
+		if(secret_key)
+			break ;
 
-        std::cerr << "Key decryption went wrong. Wrong passwd?" << std::endl;
-        last_passwd_was_wrong = true ;
-    }
-    if(!secret_key)
-    {
-        std::cerr << "Could not obtain secret key. Signature cancelled." << std::endl;
-        return false ;
-    }
+		std::cerr << "Key decryption went wrong. Wrong password?" << std::endl;
+		last_passwd_was_wrong = true ;
+	}
+	if(!secret_key)
+	{
+		std::cerr << "Could not obtain secret key. Signature cancelled." << std::endl;
+		return false ;
+	}
 
 	// then do the signature.
 
+	RsStackMutex mtx(pgphandlerMtx) ;				// lock access to PGP memory structures.
```

Comment:
better use the RS_STACK_MUTEX macro so we avoid compile complaining and better debug is possible

Reply:
Done